api/web3server.go

```diff
@@ -491,6 +496,29 @@ func (svr *Web3Server) sendRawTransaction(in interface{}) (interface{}, error) {
 		}
 	}
 
+	// extract signature from tx
+	if req.Signature, err = action.GetSignatureFromEthTX(tx, chainID); err != nil {
+		return nil, err
+	}
+
+	var rawHash []byte
+	if isEthEncoding {
+		req.Encoding = iotextypes.Encoding_ETHEREUM_RLP
+		h := types.NewEIP155Signer(big.NewInt(int64(chainID))).Hash(tx)
+		rawHash = h[:]
+	} else {
+		req.Encoding = iotextypes.Encoding_IOTEX_PROTOBUF
+		h := hash.Hash256b(byteutil.Must(proto.Marshal(req.Core)))
+		rawHash = h[:]
```

Comment:
keep the above `tx, sig, pubkey, err := action.DecodeRawTx()`
so in web3server we don't need to handle these details, they belong to action package